yarn-project/acir-simulator/src/client/client_execution_context.ts

```diff
@@ -116,40 +123,32 @@ export class ClientTxExecutionContext {
     storageSlot: ACVMField,
     sortBy: ACVMField[],
     sortOrder: ACVMField[],
-    limit: ACVMField,
-    offset: ACVMField,
-    returnSize: ACVMField,
+    limit: number,
+    offset: number,
+    returnSize: number,
   ): Promise<ACVMField[]> {
-    const { pendingCount, pendingPreimages } = this.getPendingNotes(
-      contractAddress,
-      storageSlot,
-      sortBy,
-      sortOrder,
-      limit,
-      offset,
-    );
+    const pendingNotes = this.getPendingNotes(contractAddress, storageSlot, sortBy, sortOrder, limit);
 
-    const dbLimit = +limit - pendingCount;
-    const { count: dbCount, notes: dbNotes } = await this.db.getNotes(
+    const dbNotes = await this.db.getNotes(
       contractAddress,
       fromACVMField(storageSlot),
       sortBy.map(field => +field),
       sortOrder.map(field => +field),
-      dbLimit,
-      +offset,
+      limit,
     );
-    // Noir (ACVM) expects a flattened (basically serialized) array of ACVMFields
-    const dbPreimages = dbNotes.flatMap(({ preimage }) => preimage).map(f => toACVMField(f));
+
+    // TODO: Sort again.
```

Comment:
Should we move to a transactional approach for the DB? and we drop the TX after finishing simulation